CaT/test/Filter/PredicateTest.php

```diff
@@ -243,9 +940,87 @@ public function test_fields() {
 		$f3 = $f->field("baz");
 
 		$this->assertEquals(array($f1, $f2), $f1->EQ($f2)->fields());
-		$this->assertEquals(array($f1, $f2, $f3), $f1->EQ($f2)->OR($f2->LE($f3))->fields());
+		$this->assertEquals(array($f1, $f2, $f3), $f1->EQ($f2)->_OR($f2->LE($f3))->fields());
 
-		// Implement some more variations
-		$this->assertFalse("Implement me!");
 	}
+	
+	public function test_IN() {
+
+		$f = $this->factory;
+		$i = $this->interpreter;
+
+		$int_list = $f->list_int(1,2,3);		
+		$int_list_v = $f->list_int();
+
+		$str_list = $f->list_str("a","b","c");		
+		$str_list_v = $f->list_str();
+		
+		$int_1 = $f->int(1);
+		$int_2 = $f->int(5);
+
+		$str_1 = $f->str("a");
+		$str_2 = $f->str("z");
+
+		$field_1 = $f->field("foo");
+
+		$this->assertTrue($i->interpret($int_1->IN($int_list),array()));
+		$this->assertFalse($i->interpret($int_1->IN($int_list_v),array()));
+		$this->assertFalse($i->interpret($int_2->IN($int_list),array()));
+
+		$this->assertTrue($i->interpret($str_1->IN($str_list),array()));
+		$this->assertFalse($i->interpret($str_1->IN($str_list_v),array()));
+		$this->assertFalse($i->interpret($str_2->IN($str_list),array()));
+
+		$this->assertFalse($i->interpret($int_1->IN($str_list),array()));
+		$this->assertFalse($i->interpret($str_1->IN($int_list),array()));
+		$this->assertFalse($i->interpret($str_1->IN($int_list_v),array()));
+		$this->assertFalse($i->interpret($int_2->IN($str_list_v),array()));
+
+		$this->assertTrue($i->interpret($field_1->IN($int_list),array("foo" => 1)));
+		$this->assertFalse($i->interpret($field_1->IN($int_list),array("bar" => "foo")));
+		$this->assertTrue($i->interpret($field_1->IN($str_list),array("foo" => "a")));
+	}
+
+	/**
+	* not needed atm
+	*/
+	public function test_LIKE() {
+		$this->assertTrue(true);
+	}
+
+
+	public function test_ValueList() {
+		$ls = $this->factory->list_int(1,2,3,4);
+		$this->assertInstanceOf("\\CaT\\Filter\\Predicates\\ValueList", $ls);
+		$vals = array_map(function (\CaT\Filter\Predicates\ValueInt $v) {
+					return $v->value();
+				}, $ls->values());
+
+		$ls = $this->factory->list_str("one","two","three");
+		$this->assertInstanceOf("\\CaT\\Filter\\Predicates\\ValueList", $ls);
+		$vals = array_map(function (\CaT\Filter\Predicates\ValueStr $v) {
+					return $v->value();
+				}, $ls->values());
+		$this->assertEquals(array("one","two","three"), $vals);
+
+		try {
+			$e = 0;
+			$ls = $this->factory->list_int(1,"one");
+		}
+		catch (\InvalidArgumentException $ex) {
+			$e = 1;
+		}
```

Comment:
Pattern ist hier: hinter dem Call der failen soll ein $this->assertFalse("Should have raised"). In den Exceptionhandler dann einfach nix rein.